pkg/acme/cert_request.go

```diff
@@ -11,16 +11,40 @@ import (
 	"io/ioutil"
 	"net/http"
 	"net/url"
+	"strings"
 	"sync"
 
 	"time"
 
 	"github.com/cenk/backoff"
 	"github.com/jetstack/kube-lego/pkg/kubelego_const"
+	"github.com/prometheus/client_golang/prometheus"
 	"golang.org/x/crypto/acme"
 	"golang.org/x/net/context"
 )
 
+var (
+	certReqs = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "cert_reqs_total",
+			Help: "Number of ACME certificate requests",
+		},
+		[]string{"domains"},
+	)
+	certReqErrors = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "cert_errors_total",
+			Help: "Number of ACME certificate request errors",
+		},
+		[]string{"domains"},
+	)
+)
```

Comment:
Please don't use global variables, place them as part of the KubeLego struct:

https://github.com/jetstack/kube-lego/blob/master/pkg/kubelego/type.go#L45


I suggest consolidating these two metrics into one:

```
certReqs = prometheus.NewCounterVec(
	prometheus.CounterOpts{
		Name: "cert_domain_authorizations_total",
		Help: "Number of ACME certificate domain authorizations",
	},
	[]string{"domain","state"},
)
```

And then I would get the authorization status from here:

https://github.com/jetstack/kube-lego/blob/master/pkg/acme/cert_request.go#L149
https://github.com/jetstack/kube-lego/blob/master/pkg/acme/cert_request.go#L151

```
metricDomainAuthorizations.WithLabelValues(domain, "failed").Inc()
metricDomainAuthorizations.WithLabelValues(domain, "succeed").Inc()
```